### PR TITLE
Fix unreliable frame-observer test

### PR DESCRIPTION
**Pull request by esanzgar**
Because of the debouncing of mutation observer, sometimes 50ms was not
enough to trigger the callback function (`onFrameAdded`).

For the `FrameObserver` tests, I have disabled the debouncing. It has
the benefit that the test run faster now.

### CODE REVIEW

File: src/annotator/test/frame-observer-test.js [hunk 1]
```diff
@@ -36,92 +28,98 @@ describe('annotator/frame-observer', () => {
     beforeEach(() => {
       container = document.createElement('div');
       document.body.appendChild(container);
+      sandbox.stub(console, 'warn');
+      $imports.$mock({
+        // Disable debouncing
+        'lodash.debounce': callback => callback,
+      });
+
```
Comment (lyzadanger) [inline src/annotator/test/frame-observer-test.js:36]: This seems like a straightforward approach to fixing this test issue, but made me curious if we have a convention around mocking third-party dependencies? I feel like we've discussed this before... @robertknight any thoughts?

File: src/annotator/test/frame-observer-test.js [hunk 2]
```diff
@@ -36,92 +28,98 @@ describe('annotator/frame-observer', () => {
...
     it('changing the `src` attribute triggers onFrameRemoved', async () => {
       const iframe = createAnnotatableIFrame();
-      await waitForFrameObserver();
+      await delay(0);
 
       assert.calledOnce(onFrameAdded);
       assert.calledWith(onFrameAdded, iframe);
 
-      iframe.setAttribute('src', document.location);
+      iframe.setAttribute('src', 'http://localhost:1');
```
Comment (robertknight) [inline src/annotator/test/frame-observer-test.js:122]: What is the reason for this change?

File: src/annotator/test/frame-observer-test.js [hunk 3]
```diff
@@ -36,92 +28,98 @@ describe('annotator/frame-observer', () => {
...
       container.remove();
       frameObserver.disconnect();
       sandbox.restore();
+      $imports.$restore();
     });
 
     it('triggers onFrameAdded when an annotatable iframe is added', async () => {
       let iframe = createAnnotatableIFrame();
-      await waitForFrameObserver();
+      await delay(0);
```
Comment (robertknight) [inline src/annotator/test/frame-observer-test.js:55]: Is this to let promise chains resolve?

**Review by robertknight — APPROVED** [review]
Looks good to me, though I had a question about an apparently unrelated change re. use of `document.location` in one test.

File: src/annotator/test/frame-observer-test.js (hunk 1 above)
Comment (robertknight) [reply src/annotator/test/frame-observer-test.js:36]: Mocking third-party dependencies is OK in some situations. The main risk to be aware of is that for tests where you do this, the test won't then catch situations where the dependency's API or behavior changes. TypeScript can still catch incompatible API changes in that situation. `lodash.debounce` has a very simple API so I think this is safe to do.

File: src/annotator/test/frame-observer-test.js (hunk 2 above)
Comment (esanzgar) [reply src/annotator/test/frame-observer-test.js:122]: I had two reasons:

* it should probably be `document.location.href`, but because `document.location` is coerced to a string it gives the same result: `document.location.toString() === document.location.href`

* it is lighter: there is a lot of JS code in that is served in the Karma `document.location` URL, that the browser will load. I just want to exercise the unloading of an iframe, so pointing to some non reachable URL, it just enough.

File: src/annotator/test/frame-observer-test.js (hunk 3 above)
Comment (esanzgar) [reply src/annotator/test/frame-observer-test.js:55]: Correct.

File: src/annotator/test/frame-observer-test.js (hunk 1 above)
Comment (esanzgar) [reply src/annotator/test/frame-observer-test.js:36]: `lodash.debounce` is mocked in several places in our code.
